src/Log/LogService.php

```diff
@@ -0,0 +1,209 @@
+ * the University Library of Hamburg University of Technology with funding from
+ * the German Research Foundation and the European Regional Development Fund.
+ *
+ * LICENCE
+ * OPUS is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the Licence, or any later version.
+ * OPUS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details. You should have received a copy of the GNU General Public License
+ * along with OPUS; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * @category    Framework
+ * @package     Opus_Bootstrap
+ * @author      Kaustabh Barman <barman@zib.de>
+ * @copyright   Copyright (c) 2008-2020, OPUS 4 development team
+ * @license     http://www.gnu.org/licenses/gpl.html General Public License
+ */
+
+namespace Opus\Log;
+
+/**
+ * Class to manage multiple loggers
+ *
+ * @package     Opus_Log
+ *
+ */
+class LogService
+{
+
+//DONE add comments for all variable -- use @var
+	/** @var Zend_Log $loggers */ 
+    public $loggers = [];		//all log objects with lognames as keys	
+    private $logFile;		//file pointer resource to the logfile
+    public $format;		//format of logging
+    private $config;	//Zend_Config
+    private $logFilePath = null;	//path to the logfile
+    private static $instance;	//log instance of the class
+    public $logFileName;	//name of the log file
+
+    public static function getInstance()
+    {
+        if (null === static::$instance) {
+            static::$instance = new static();
+        }
+        return static::$instance;
+    }
+
+    public function openLog($logFileName = null)		//change to createLog('default', logFileName)
+    {
+        $this->getConfig();
+
+        $logfilePath = $this->logFilePath == null ? $this->setPath($logFileName) : $this->logFilePath;
+
+        if ($logFileName == null) {
+        	$this->logFileName = 'opus.log'
+        } else {
+        	$this->logFileName = $logFileName;
+        }
+        $this->logFile = @fopen($logfilePath, 'a', false);
+        if ($this->logFile === false) {
+            $path = dirname($logfilePath);
+
+            if (! is_dir($path)) {
+                throw new Exception('Directory for logging does not exist');
+            } else {
+                throw new Exception('Failed to open logging file:' . $logfilePath);
+            }
+        }
+    }
+
+    public function getConfig()		//DONE not going to work
+    {
+        $this->config = \Zend_Registry::get('Zend_Config');
+    }
+
+    public function getPath()
+    {
+        return $this->logFilePath;
+    }
+
+    public function setPath($logFileName, $logPath = null)
```

Comment:
It would be good to have /** ... */ comments for these functions with a description of the intend. I don't think we need this function the way it is, but maybe I don't see the idea.

Reply:
This function is to change the path of the logfile. It will be useful for running tests. I think you mentioned this on my previous commit.